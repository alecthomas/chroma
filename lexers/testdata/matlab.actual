% a sum function
function p = vector_sum(x)
p = 0;
for k1 = 1:length(x)
    p = p + x(k1);
end
end

z = 1:10;
sum_from_1_to_10 = vector_sum(z)
