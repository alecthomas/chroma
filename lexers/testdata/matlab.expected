[
  {"type":"Comment","value":"% a sum function"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"function"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"p "},
  {"type":"Punctuation","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameFunction","value":"vector_sum"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":"x"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Name","value":"p"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"0"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"for"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"k1"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":":"},
  {"type":"NameBuiltin","value":"length"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"x"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":"\n    "},
  {"type":"Name","value":"p"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"p"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"x"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"k1"},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"end"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"end"},
  {"type":"Text","value":"\n\n"},
  {"type":"Name","value":"z"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":":"},
  {"type":"LiteralNumberInteger","value":"10"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Name","value":"sum_from_1_to_10"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"vector_sum"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"z"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":"\n"}
]
